### PR TITLE
Do not create gift cards for waiting for approval orders

### PR DESCRIPTION
**Pull request by IKarbowiak**
When an order is waiting for approval, the fulfillments lines are created in the background. In this situation, the gift card shouldn't be created.
Gift cards should be created when fulfillment is approved.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-4509 Do not create gift card for waiting for approval orders](https://app.clickup.com/t/2549495/SALEOR-4509)

**Review by maarcingebala — APPROVED** [review]

File: CHANGELOG.md
Comment (maarcingebala) [inline CHANGELOG.md:15]: I'm not sure we have to update changelog in this case - this is part of the gift cards feature that will be introduced in 3.1 and we don't need to describe every change. I will just include this in release notes but I will know about this PR from the list of merged PRs.

Comment (IKarbowiak) [reply CHANGELOG.md:15]: Ok. I'll remove that.

File: saleor/graphql/order/mutations/fulfillments.py
Comment (tomaszszymanski129) [inline saleor/graphql/order/mutations/fulfillments.py:230]: I don't get it

File: saleor/graphql/order/mutations/fulfillments.py
Comment (tomaszszymanski129) [inline saleor/graphql/order/mutations/fulfillments.py:229]: that method returns `get_gift_card_lines(pks)`, isn't that endless loop?

**Review by tomaszszymanski129 — APPROVED** [review]
What about this method `get_gift_card_lines`? I see we don't make a use of it in this PR, is it intentional? Otherwise looks good.

File: saleor/graphql/order/mutations/fulfillments.py
Comment (IKarbowiak) [reply saleor/graphql/order/mutations/fulfillments.py:230]: 😮

**Review by eCzerniak — APPROVED** [review]

File: saleor/graphql/order/mutations/fulfillments.py
Comment (tomaszszymanski129) [inline saleor/graphql/order/mutations/fulfillments.py:227]: we could just `get_gift_card_lines(order_lines)` and drop method from line 233. It usually seems a good idea to extract such things to class methods but since this is another function anyway I wouldn't mind to drop it, to keep the class bit more compact.

**Review by tomaszszymanski129 — APPROVED** [review]
What about this method `get_gift_card_lines`? I see we don't make a use of it in this PR, is it intentional? Otherwise looks good.

File: saleor/graphql/order/mutations/fulfillments.py
Comment (IKarbowiak) [reply saleor/graphql/order/mutations/fulfillments.py:227]: True. :)

**Pull request merged**
